src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -277,6 +328,16 @@ public unsafe int Seek(ref Vector<byte> byte0Vector)
                     var pEnd = pCurrent + following;
                     do
                     {
+                        bytesScanned++;
+
+                        if ((bytesScanned > limitBytes) ||
+                            (!limitIterator.IsDefault && block == limitIterator.Block && index > limitIterator.Index))
```

Comment:
instead of doing this complicated comparison for every byte, couldn't you adjust `pEnd` to match the limit index?